### PR TITLE
feat: Add SKCanvasView

### PR DESCRIPTION
**Pull request by workgroupengineering**
A Xaml canvas control that can be drawn on using SkiaSharp drawing commands
which facilitates porting from existing `Xamarin Forms` applications.

### CODE REVIEW

**Comment by workgroupengineering** [comment]
> Wouldn't it be better to provide hw-acceleration when it's available?

Yes. Can you tell me where to start doing?

**Comment by kekekeks** [comment]
The easiest way would be to wrap OpenGlControlBase since it provides a Skia-compatible context

**Comment by workgroupengineering** [comment]
> The easiest way would be to wrap OpenGlControlBase since it provides a Skia-compatible context

Thanks, I'll try.

File: src/Avalonia.Labs.Controls/SKCanvasView/SKPaintSurfaceEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+namespace Avalonia.Labs.Controls;
+
+public sealed class SKPaintSurfaceEventArgs : EventArgs
+{
+    public SKPaintSurfaceEventArgs(SKSurface surface, SKImageInfo info)
+        : this(surface, info, info)
+    {
+    }
+
+    public SKPaintSurfaceEventArgs(SKSurface surface, SKImageInfo info, SKImageInfo rawInfo)
```
Comment (kekekeks) [inline src/Avalonia.Labs.Controls/SKCanvasView/SKPaintSurfaceEventArgs.cs:13]: IMO we should provide an SKCanvas instance there and make the rest to be optional.
We might want to have a mode that utilizes SKPictureRecorder and custom composition visual to make SKCanvasView to be hardware-accelerated even on platforms where we don't provide GPU surface interop.

**Review by kekekeks — CHANGES_REQUESTED** [review]

**Comment by kekekeks** [comment]
I guess the GPU support should be a separate PR, address review comments and let's get it merged

File: src/Avalonia.Labs.Controls/SKCanvasView/SKPaintSurfaceEventArgs.cs (hunk 1 above)
Comment (workgroupengineering) [reply src/Avalonia.Labs.Controls/SKCanvasView/SKPaintSurfaceEventArgs.cs:13]: I would like to maintain compatibility with https://github.com/mono/SkiaSharp/blob/25e70a390e2128e5a54d28795365bf9fdaa7161c/source/SkiaSharp.Views/SkiaSharp.Views.Shared/SKPaintSurfaceEventArgs.cs for quick porting.

**Review by maxkatz6 — APPROVED** [review]

**Comment by robloo** [comment]
Note: This seems to be based on the code provided in the discussion here: https://github.com/AvaloniaUI/Avalonia/discussions/12269. That code is itself a rough port from SkiaSharp itself.

 1) As mentioned that code was really rough and I really didn't intend for it to get pulled in as-is. Otherwise, I would have done it myself.
 1) It might be a good idea to mention this code is a port from SkiaSharp's UWP control
 2) It might also be a good idea at some point to get this control in upstream instead

This kind of detail is always great in the PR description for future reference.

**Pull request merged**